src/main/java/org/broadinstitute/consent/http/service/DatabaseMatchingServiceAPI.java

```diff
@@ -69,17 +69,20 @@ public Match findSingleMatch(String consentId, String purposeId) {
 
     @Override
     public Match findMatchForPurpose(String purposeId){
-        Document dar = dataAccessAPI.describeDataAccessRequestById(purposeId);
-        Consent consent = findRelatedConsents(purposeId);
         Match match = null;
-        if(dar != null){
+        Document dar = dataAccessAPI.describeDataAccessRequestById(purposeId);
+        if (Objects.nonNull(dar)) {
+            List<Integer> dataSetIdList = DarUtil.getIntegerList(dar, DarConstants.DATASET_ID);
+            Consent consent = findRelatedConsent(dataSetIdList);
+            if (Objects.nonNull(consent)) {
```

Comment:
Null checks necessary to catch a bug where this would fail and the DAR creation process was interrupted. Failures here are not an exception and should not stop DAR creation.